lib/Graph/Graph.cpp

```diff
@@ -474,7 +475,12 @@ PadNode *Function::createPad(llvm::StringRef name, NodeValue input,
                              llvm::ArrayRef<int> pads, float value) {
   assertPadsSize(input, pads);
   auto OT = getParent()->uniqueType(*outTy);
-  return addNode(new PadNode(name, OT, input, mode, pads, value));
+  PadNode *newNode = addNode(new PadNode(name, OT, input, mode, pads, value));
+
+  // Log creation of node
+  LOG_NODE_CREATE_WITH_ONE_INPUT(newNode, input.getNode())
```

Comment:
Is is possible to avoid touching all those `create` functions? 

May be one could intercept e.g. `Node::Node` constructor and do something there? This constructor would have the `Node *` as `this` and it would know the `name` of the new Node.  But I don't know how to intercept `input` there...

Alternatively, you could try to change `NodeGen` so that it emits some code into constructors of auto-generated nodes to call  a matching `LOG_NODE_CREATE_WITH_XYZ`.

Reply:
Done by moving the logging inside F->addNode()